spring-cloud-starter-task-composedtaskrunner/src/main/java/org/springframework/cloud/task/app/composedtaskrunner/ComposedRunnerJobBuilder.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.app.composedtaskrunner;
+
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.springframework.batch.core.Step;
+import org.springframework.batch.core.job.builder.FlowBuilder;
+import org.springframework.batch.core.job.flow.Flow;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.dataflow.core.dsl.FlowNode;
+import org.springframework.cloud.dataflow.core.dsl.LabelledTaskNode;
+import org.springframework.cloud.dataflow.core.dsl.SplitNode;
+import org.springframework.cloud.dataflow.core.dsl.TaskAppNode;
+import org.springframework.cloud.dataflow.core.dsl.TaskParser;
+import org.springframework.cloud.dataflow.core.dsl.TransitionNode;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.util.Assert;
+
+/**
+ * Genererates a Composed Task Job Flow.
+ *
+ * @author Glenn Renfro
+ */
+public class ComposedRunnerJobBuilder {
```

Comment:
Why is this not a `FactoryBean` that encapsulates the parsing and generation?  The way this is configured in `ComposedTaskRunnerConfiguration` is odd...the `ComposedRunnerVisitor` and this "Builder" are beans.  It feels like the encapsulation of what's going on is wrong.  I'd expect you pass the DSL to a `FactoryBean` and out comes the job.  Instead it's really a collection of beans that all are tightly coupled to each other.